tests/cilium_network_policy_test.cc

```diff
@@ -1955,5 +1948,108 @@ TEST_F(CiliumNetworkPolicyTest, SNIPatternMatching) {
   EXPECT_FALSE(wildcard_label.matches(""));
 }
 
+TEST_F(CiliumNetworkPolicyTest, OrderedRules) {
```

Comment:
Maybe we need a test for multiple rules and one of the rules that has precedence not specified?